ironfish/src/rpc/routes/wallet/getAccountTransactionDescriptions.ts

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { UnsignedTransaction } from '../../../primitives/unsignedTransaction'
+import { ApiNamespace } from '../namespaces'
+import { routes } from '../router'
+import { AssertHasRpcContext } from '../rpcContext'
+import { getAccount } from './utils'
+
+export type GetAccountTransactionDescriptionsRequest = {
+  unsignedTransaction: string
+  account?: string
+}
+
+export type GetAccountTransactionDescriptionsResponse = {
+  account: string
+  mints: {
+    assetId: string
+    value: string
+    transferOwnershipTo?: string
+  }[]
+  burns: {
+    assetId: string
+    value: string
+  }[]
+  receivedNotes: {
+    assetId: string
+    memo: string
+    owner: string
+    sender: string
+    value: string
+  }[]
+  sentNotes: {
+    assetId: string
+    memo: string
+    owner: string
+    sender: string
+    value: string
+  }[]
+}
+
+export const GetAccountTransactionDescriptionsRequestSchema: yup.ObjectSchema<GetAccountTransactionDescriptionsRequest> =
```

Comment:
This isn't getting the descriptions, this is decrypting the unsigned transaction. Technically I think it should be able to take in a signed or unsigned transaction as well.

Reply:
Aren't descriptions supposed to be the publically viewable contents of a transaction? For mints/burns they just happen to be public in general. The notes are private

Reply:
It's a mix of both I guess. It's returning the mint/burn descriptions but also decrypting the notes associated with the output descriptions on the unsigned transaction.

The primary use case here is just to be for the unsigned transaction to show more details. But if you think it's useful to expand this, I can rework it or rename it so it's more explicit for unsigned transactions.